### ISSUE TITLE
Calling clear() throws IllegalArgumentException

### ISSUE THREAD

**Issue by rishabh9** [OpenHFT/SmoothieMap#6]
On calling `clear()` the below exception is thrown:
```
Exception in thread "main" java.lang.IllegalArgumentException
	at java.base/jdk.internal.misc.Unsafe.invalidInput(Unsafe.java:437)
	at java.base/jdk.internal.misc.Unsafe.checkPrimitiveArray(Unsafe.java:557)
	at java.base/jdk.internal.misc.Unsafe.checkPrimitivePointer(Unsafe.java:576)
	at java.base/jdk.internal.misc.Unsafe.setMemoryChecks(Unsafe.java:747)
	at java.base/jdk.internal.misc.Unsafe.setMemory(Unsafe.java:718)
	at jdk.unsupported/sun.misc.Unsafe.setMemory(Unsafe.java:527)
	at net.openhft.smoothie.Segment.clear(Segment.java:507)
	at net.openhft.smoothie.SmoothieMap.clear(SmoothieMap.java:1244)
	at Main.main(Main.java:11)
```
The below code can easily reproduce the issue. I am using OpenJDK 9.0.4.

```
import net.openhft.smoothie.SmoothieMap;

import java.util.Map;
import java.util.Optional;

public class Main {

    public static void main(String[] args) {
        Map<String, Optional<String>> map = new SmoothieMap<>(10);
        map.put("Hello", Optional.of("World!"));
        map.clear();
    }
}
```
Could someone please take a look into this?

**Comment by leventov** [comment]
Thanks for this report, interesting. Actually I'm in the process of rewriting SmoothieMap at the moment so I won't release any fix quickly, but if you want to to continue experimenting with SmoothieMap, you could replace that `setMemory()` call in `Segment.java:507` with a loop:
```java
for (long offset = HASH_TABLE_OFFSET; offset < HASH_TABLE_OFFSET + HASH_TABLE_SIZE * HASH_TABLE_SLOT_SIZE; offset += Long.BYTES) {
    U.putLong(this, offset, 0L);
}
```

**Comment by rishabh9** [comment]
Thank you for the information. I'll give that a try. Shall we leave the
ticket open? This information will help others.

**Comment by leventov** [comment]
This should now be fixed in SmoothieMap 2.0.1
